document/core/exec/modules.rst

```diff
@@ -371,6 +371,66 @@ New instances of :ref:`functions <syntax-funcinst>`, :ref:`tables <syntax-tablei
    \end{array}
 
 
+.. index:: element, element instance, element address
+.. _alloc-elem:
+
+:ref:`Element segments <syntax-eleminst>`
+.........................................
+
+1. Let :math:`e` be the :ref:`element segment <syntax-elem>` to allocate.
+
+2. If :math:`e` is of the form :math:`\{ \EINIT~x^\ast \}`, then:
+
+   a. Let :math:`a` be the first free :ref:`element address <syntax-elemaddr>` in :math:`S`.
+
+   b. For each :ref:`function index <syntax-funcidx>` :math:`x_i` in :math:`e.\EINIT`, do:
+
+      i. Let :math:`\funcaddr_i` be the :ref:`function address <syntax-funcaddr>` :math:`\moduleinst.\MIFUNCS[x_i]`.
+
+   c. Let :math:`\funcaddr^\ast` be the concatenation of the function addresses :math:`\funcaddr_i`.
+
+   d. Let :math:`\eleminst` be the :ref:`element instance <syntax-eleminst>` :math:`\{ \ESIINIT~\funcaddr^\ast \}`.
+
+   e. Append :math:`\eleminst` to the |SELEM| of :math:`S`.
+
+.. math::
+   \begin{array}{rlll@{\qquad}l}
+   \allocelem(S, e, \moduleinst) &=& S', \elemaddr & (\iff e = \{ \EINIT~x^\ast \}) \\[1ex]
+   \allocelem(S, e, \moduleinst) &=& S & (\otherwise) \\[1ex]
```

Comment:
This only returns one thing, not a pair, which seems incoherent. I think it should be epsilon.